packages/store/package.json

```diff
@@ -15,10 +15,11 @@
   },
   "scripts": {
     "prepare": "yarn build",
-    "codegen": "ts-node ./scripts/codegen.ts",
+    "codegen": "yarn tsgen && yarn tablegen",
+    "tsgen": "ts-node ./scripts/codegen.ts",
     "tablegen": "../cli/dist/mud.js tablegen",
     "test": "yarn codegen && forge test",
-    "build": "yarn codegen && rimraf out && forge build && yarn dist && yarn types && tsup",
+    "build": "rimraf out && forge build && yarn dist && yarn types && tsup",
```

Comment:
Using `tablegen` here caused an error in the CI, because `../cli/dist/mud.js` does not exist yet while store is being built (CLI can't be built first because it depends on `World`'s ABI, which depends on `Store`). There are (at least) two approaches to solve this:
1. Move the `tablegen` scripts into `Store` (and `worldgen` scripts into `World`), so these packages can use them locally (and consumers of these packages don't have to install the MUD CLI if they only want to use eg `Store` without the rest of the framework). CLI can still expose a `tablegen` command by importing the scripts from `Store` (and a `worldgen` command by importing from `World`)
2. Check in the World/Store ABIs, so CLI can be built independently from building World/Store (this alone doesn't fix the circular dependency though)

Reply:
since this has come up a bunch, mind opening an issue for this?

Reply:
https://github.com/latticexyz/mud/issues/616